### PR TITLE
gautrigger_sync

### PR DESCRIPTION
**Pull request by alexey-vyrskiy**
@sardeie pull request has been created.
Could you please check sources?
And I need about an hour for additional check and a little dev testing.

### CODE REVIEW

File: force-app/main/default/objects/Allocation_To_OppPurpose_Field_Mapping__mdt/Allocation_To_OppPurpose_Field_Mapping__mdt.object-meta.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sardeie) [inline force-app/main/default/objects/Allocation_To_OppPurpose_Field_Mapping__mdt/Allocation_To_OppPurpose_Field_Mapping__mdt.object-meta.xml:1]: Did you create this to make the solution admin-configurable? If so, I would prefer to make this dev-controlled only, so please get rid of this custom metadata type and hard code the fields.

If there is another reason for it, please let me know.

File: force-app/main/default/objects/Opportunity_Purpose__c/fields/Unique_Allocation_Id__c.field-meta.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sardeie) [inline force-app/main/default/objects/Opportunity_Purpose__c/fields/Unique_Allocation_Id__c.field-meta.xml:1]: Client can create multiple Opportunity Purpose records for a single allocation. In our automation, we want to maintain a one-to-one mapping, but we don't want to enforce it so client has flexibility to make changes later.

File: force-app/main/default/flows/Unique_Allocation_Id_Filling.flow-meta.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sardeie) [inline force-app/main/default/flows/Unique_Allocation_Id_Filling.flow-meta.xml:1]: Let's delete this. We don't want the enforce a unique one-to-one constraint.

File: force-app/main/default/classes/AllocationTriggerHandler.cls [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public override void afterInsert() {
+    	AllocationToOppPurposeSyncHelper.createRelatedOpportunityPurposes(newList);
+    }
+
+    public override void afterUpdate() {
+    	AllocationToOppPurposeSyncHelper.updateRelatedOpportunityPurposes(newList, oldMap);
+    }
+    
+    public override void beforeDelete() {
+    	AllocationToOppPurposeSyncHelper.collectRelatedOpportunityPurposesforDelete(oldMap);
```
Comment (sardeie) [inline force-app/main/default/classes/AllocationTriggerHandler.cls:26]: Does this work? Are you sure static properties set in **before delete** will be available in **after delete**?

**Review by sardeie — CHANGES_REQUESTED** [review]
Thank you Alexey! Please see the requested changes in the comments.

File: force-app/main/default/objects/Allocation_To_OppPurpose_Field_Mapping__mdt/Allocation_To_OppPurpose_Field_Mapping__mdt.object-meta.xml (hunk 1 above)
Comment (alexey-vyrskiy) [reply force-app/main/default/objects/Allocation_To_OppPurpose_Field_Mapping__mdt/Allocation_To_OppPurpose_Field_Mapping__mdt.object-meta.xml]: custom metadata mapping supposes only configuration without deploy.
Deleted
All fields hard coded  in Helper

File: force-app/main/default/objects/Opportunity_Purpose__c/fields/Unique_Allocation_Id__c.field-meta.xml (hunk 2 above)
Comment (alexey-vyrskiy) [reply force-app/main/default/objects/Opportunity_Purpose__c/fields/Unique_Allocation_Id__c.field-meta.xml]: fixed

File: force-app/main/default/flows/Unique_Allocation_Id_Filling.flow-meta.xml (hunk 3 above)
Comment (alexey-vyrskiy) [reply force-app/main/default/flows/Unique_Allocation_Id_Filling.flow-meta.xml]: deleted

File: force-app/main/default/classes/AllocationTriggerHandler.cls (hunk 4 above)
Comment (alexey-vyrskiy) [reply force-app/main/default/classes/AllocationTriggerHandler.cls:26]: Yes, it definitely works and passed in test class.

**Comment by sardeie** [comment]
Thank you @alexey-vyrskiy. Looks good. I'm merging it.

**Pull request merged**